### PR TITLE
[AIRFLOW-5475] Normalize gcp_conn_id in operators and hooks

### PR DESCRIPTION
**Pull request by TobKed**
Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-5475
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

File: UPDATING.md [hunk 1]
```diff
@@ -40,6 +40,13 @@ assists users migrating to a new version.
 
 ## Airflow Master
 
+### Normalize gcp_conn_id for Google Cloud Platform
+
+Previously not all hooks and operators related to Google Cloud Platform use 
+conn_id as parameter for GCP connection. There is currently one parameter 
+which apply to most services. Parameters like ``datastore_conn_id``, ``bigquery_conn_id``,
+``google_cloud_storage_conn_id`` have been deprecated.
```
Comment (mik-laj) [inline UPDATING.md]: Can you list all affected operators here? In this PR and my previour PR that make similar changes.

Comment (TobKed) [reply UPDATING.md]: Sure. I've added list of components.

File: airflow/gcp/hooks/bigquery.py [hunk 2]
```diff
@@ -47,15 +48,23 @@ class BigQueryHook(GoogleCloudBaseHook, DbApiHook):
...
                  use_legacy_sql: bool = True,
-                 location: Optional[str] = None) -> None:
+                 location: Optional[str] = None,
+                 bigquery_conn_id: Optional[str] = None) -> None:
+        # To preserve backward compatibility
+        # TODO: remove one day
+        if bigquery_conn_id:
+            warnings.warn(
+                "The bigquery_conn_id parameter has been deprecated. You should pass "
+                "the gcp_conn_id parameter.", DeprecationWarning)
```
Comment (mik-laj) [inline airflow/gcp/hooks/bigquery.py]: isn't stack level missing here? For me it should be 2.

File: airflow/gcp/hooks/datastore.py [hunk 3]
```diff
@@ -40,11 +41,19 @@ class DatastoreHook(GoogleCloudBaseHook):
...
-        super().__init__(datastore_conn_id, delegate_to)
+    def __init__(
+        self,
+        gcp_conn_id: str = 'google_cloud_default',
+        delegate_to: Optional[str] = None,
+        api_version: str = 'v1',
+        datastore_conn_id: Optional[str] = None
+    ) -> None:
+        if datastore_conn_id:
+            warnings.warn(
```
Comment (mik-laj) [inline airflow/gcp/hooks/datastore.py:52]: Isn't stack level missing here?

File: airflow/gcp/hooks/gcs.py [hunk 4]
```diff
@@ -42,11 +43,20 @@ class GoogleCloudStorageHook(GoogleCloudBaseHook):
...
+        self,
+            gcp_conn_id: str = 'google_cloud_default',
+            delegate_to: Optional[str] = None,
+            google_cloud_storage_conn_id: Optional[str] = None
+    ) -> None:
+        # To preserve backward compatibility
+        # TODO: remove one day
+        if google_cloud_storage_conn_id:
+            warnings.warn(
+                "The google_cloud_storage_conn_id parameter has been deprecated. You should pass "
```
Comment (mik-laj) [inline airflow/gcp/hooks/gcs.py:57]: Isn't stack level missing here?

**Review by mik-laj — CHANGES_REQUESTED** [review]

File: UPDATING.md (hunk 1 above)
Comment (TobKed) [reply UPDATING.md]: I've fixed this list. Previously paths were not complete (missing file name).

File: airflow/gcp/hooks/gcs.py (hunk 4 above)
Comment (TobKed) [reply airflow/gcp/hooks/gcs.py:57]: Stacklevel info added.

File: airflow/gcp/hooks/datastore.py (hunk 3 above)
Comment (TobKed) [reply airflow/gcp/hooks/datastore.py:52]: Stacklevel info added.

File: airflow/gcp/hooks/bigquery.py (hunk 2 above)
Comment (TobKed) [reply airflow/gcp/hooks/bigquery.py]: Stacklevel info added.

File: UPDATING.md [hunk 5]
```diff
@@ -40,6 +40,40 @@ assists users migrating to a new version.
 
 ## Airflow Master
 
+### Normalize gcp_conn_id for Google Cloud Platform
+
+Previously not all hooks and operators related to Google Cloud Platform use 
+conn_id as parameter for GCP connection. There is currently one parameter
```
Comment (mik-laj) [inline UPDATING.md]:
```suggestion
`gcp_conn_id` as parameter for GCP connection. There is currently one parameter 
```

File: UPDATING.md [hunk 6]
```diff
@@ -40,6 +40,40 @@ assists users migrating to a new version.
 
 ## Airflow Master
 
+### Normalize gcp_conn_id for Google Cloud Platform
+
+Previously not all hooks and operators related to Google Cloud Platform use 
+conn_id as parameter for GCP connection. There is currently one parameter 
+which apply to most services. Parameters like ``datastore_conn_id``, ``bigquery_conn_id``,
+``google_cloud_storage_conn_id`` and similar have been deprecated.
```
Comment (mik-laj) [inline UPDATING.md]:
```suggestion
``google_cloud_storage_conn_id`` and similar have been deprecated. Operators that require two connections are not changed.
```

**Comment by TobKed** [comment]
cc @mik-laj

**Review by mik-laj — APPROVED** [review]

**Pull request merged**